src/mailmojo-widget.php

```diff
@@ -48,22 +54,54 @@ public function initFiles () {
 	}
 
 	/**
+	 * Return true if the widget is enabled.
+	 *
+	 * Minimum widget data needs to be present: subscribeurl or the deprecated listid.
+	 *
+	 * @return bool
+	 */
+	public function isEnabled () {
+		$options = get_option('mailmojo_options');
+		// TODO: Remove options check on listid when username is removed
+		return !empty($options['subscribeurl']) || !empty($options['listid']);
+	}
+
+	/**
 	 * Outputs the widget form on widgets admin page.
 	 *
 	 * @param array $instance
 	 */
 	public function form ($instance) {
-		if (!$this->plugin->isActive()) {
-			$url = $this->plugin->getSettingsPageUrl();
+		$settingsPageUrl = $this->plugin->getSettingsPageUrl();
 
+		if (!$this->plugin->isActive()) {
 			echo '<p>' . sprintf(
 				__('You need to enter your MailMojo account information on the <a href="%s">MailMojo settings page</a>', 'mailmojo'),
-				$url) . '</p>';
+					 $settingsPageUrl) . '</p>';
 			return;
 		}
 
+		if (self::$lists === null) {
+			try {
+				$api = new MailMojo\Api\ListsApi();
+				self::$lists = $api->getLists();
+			}
+			catch (MailMojo\ApiException $e) {
+				echo '<p>' . sprintf(
+					__('The access token is not working correctly. Go to the <a href="%s">MailMojo settings page</a> and make sure the access token is correct.', 'mailmojo'),
```

Comment:
A bit more user friendly IMO:

"Your email lists could not be retrieved. Please go to … and make sure the access token is correct."